src/transformers/generation/utils.py

```diff
@@ -1594,6 +1592,23 @@ def generate(
                 )
             if generation_config.cache_implementation == "static":
                 model_kwargs["past_key_values"] = self._get_static_cache(batch_size, generation_config.max_length)
+            elif generation_config.cache_implementation == "quantized":
+                if not is_quanto_available():
+                    raise ImportError(
+                        "You need to install `quanto` in order to use KV cache quantization. "
+                        "Please install it via  with `pip install git+https://github.com/huggingface/quanto`"
+                    )
+
+                cache_config = (
+                    generation_config.cache_config
+                    if generation_config.cache_config is not None
+                    else QuantizedCacheConfig()
+                )
+                model_kwargs["past_key_values"] = QuantoQuantizedCache(
```

Comment:
Here we should only give the cache config, not the kwargs

Reply:
But we need to give user freedom in changing number of bits, group size and other params while quantizing. I do not see other places where we can set those kwargs for the Cache Class

Reply:
Ahh you mean changing these from the kwargs of generate? Makes sense, but should be handled the same way we handle the generation config maybe?

Reply:
I decided to pass cache config as is to the `QuantoQuantizedCache`, without explicitly indicating each kwarg

Reply:
thanks!